src/drivers/block_dev/ramdisk/ramdisk.c

```diff
@@ -167,7 +166,7 @@ static int read_sectors(block_dev_t *bdev,
 	char *read_addr;
 
 	ramdisk = (ramdisk_t *) bdev->privdata;
-	read_addr = ramdisk->p_start_addr + (blkno * ramdisk->block_size);
+	read_addr = ramdisk->p_start_addr + (blkno * bdev->block_size);
```

Comment:
It seems something is wrong with ramdisk->block_size. If I understand right ramdisk->block_size and bdev->block_size mean the same thing. So may be remove the first one at all?